onadata/apps/logger/models/xform_submissions_counter.py

```diff
@@ -0,0 +1,25 @@
+import datetime
+
+from django.contrib.auth.models import User
+from django.contrib.gis.db import models
+from django.utils import timezone
+
+from onadata.apps.logger.models.xform import XForm
+
+
+class XFormSubmissionCounter(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    xform = models.ForeignKey(XForm, related_name='xformsubmissioncounter', on_delete=models.CASCADE)
```

Comment:
we actually want to retain counts for deleted projects. we'll have to think about this more.

Reply:
let's create a constant somewhere, like `DELETED_XFORM_PK = -1`, and assign `xform_id` to that value for counter instances whose linked `XForm` has been deleted.

that then creates the problem of needing to have an `XForm` in the database whose primary key matches that value :smiling_imp: there's an example of something similar for anonymous users: https://github.com/kobotoolbox/kpi/blob/288e5e5e8b105df663267254c59304c8c946d785/kpi/utils/object_permission.py#L65. you could follow this example and create a utility function that returns the "ghost" `XForm` instance that serves as a placeholder for deleted forms, having the utility create that instance in the database if needed.